### DISCUSSION TITLE
How to find out the number of files in a specified path in Storage?

### DISCUSSION THREAD

**Discussion by JCtapuk** [supabase/supabase#14917] [Questions] [answered]
I searched everywhere and did not find it. Apparently I need to request a storage scheme through databases?

**Comment by GaryAustin1** [comment, marked as answer]
You could use .list() https://supabase.com/docs/reference/javascript/storage-from-list which takes a path and get the length of the array.   Otherwise you can do SQL on the storage.objects table with an rpc function or with a client set to access storage schema (assuming you meet RLS).

**Reply by JCtapuk** [reply]
> You could use .list() https://supabase.com/docs/reference/javascript/storage-from-list which takes a path and get the length of the array. Otherwise you can do SQL on the storage.objects table with an rpc function or with a client set to access storage schema (assuming you meet RLS).

I know that this is a sheet, it returns the entire list for the file limit, and I want to paginate from it, I have 10,000 pictures and display and count everything, and then pagination is not recommended. for example, a query for the base count (*) I need such an analogue for storage

**Reply by JCtapuk** [reply]
they would simply lead a function and not a perverted scheme to create a new client that is available to all folders
storage.from('assets').count(<path>): return number

**Reply by GaryAustin1** [reply]
I would approach it with an rpc call to a Postgres function that selects count from storage.objects doing a WHERE clause based on the array in the column path_tokens which has every "folder" in the path in order and bucket_id(name).   You could also string search the name column based on your path and bucket name.  

You might be able to do this from the API by creating a client with the schema option set to storage and then the appropriate filters and return count.

**Reply by JCtapuk** [reply]
> I would approach it with an rpc call to a Postgres function that selects count from storage.objects doing a WHERE clause based on the array in the column path_tokens which has every "folder" in the path in order and bucket_id(name). You could also string search the name column based on your path and bucket name.
> 
> You might be able to do this from the API by creating a client with the schema option set to storage and then the appropriate filters and return count.

well, I don’t like this approach, why not just add a new function to storage that returns the number of files in the specified path rather than creating 2 clients perversely. I'll try via rpc if it's available for me in the free version

**Reply by GaryAustin1** [reply]
Sorry, but there is nothing built in to do this that I'm aware of.  You can certainly suggest to Supabase either in github supabase/storage-api or with feature request here.  There is no guarantee they will read your comment here.

 rpc is just an API call to access a Postgres function and you can pass in parameters. And is available on free plans.  I would not go the 2nd client path either, but just giving ideas I could think of as work a rounds.

**Reply by JCtapuk** [reply]
> Sorry, but there is nothing built in to do this that I'm aware of. You can certainly suggest to Supabase either in github supabase/storage-api or with feature request here. There is no guarantee they will read your comment here.
> 
> rpc is just an API call to access a Postgres function and you can pass in parameters. And is available on free plans. I would not go the 2nd client path either, but just giving ideas I could think of as work a rounds.

It seems that it turned out to create an rpc function.
